js/woo.js

```diff
@@ -0,0 +1,1586 @@
+        //
+        // {
+        //     success: true,
+        //     symbol: 'SPOT_WOO_USDT',
+        //     status: 'FILLED', // FILLED, NEW
+        //     side: 'BUY',
+        //     created_time: '1641480933.000',
+        //     order_id: '87541111',
+        //     order_tag: 'default',
+        //     price: '1',
+        //     type: 'LIMIT',
+        //     quantity: '12',
+        //     amount: null,
+        //     visible: '12',
+        //     executed: '12', // or any partial amount
+        //     total_fee: '0.0024',
+        //     fee_asset: 'WOO',
+        //     client_order_id: null,
+        //     average_executed_price: '1',
+        //     Transactions: [
+        //       {
+        //         id: '99111647',
+        //         symbol: 'SPOT_WOO_USDT',
+        //         fee: '0.0024',
+        //         side: 'BUY',
+        //         executed_timestamp: '1641482113.084',
+        //         order_id: '87541111',
+        //         executed_price: '1',
+        //         executed_quantity: '12',
+        //         fee_asset: 'WOO',
+        //         is_maker: '1'
+        //       }
+        //     ]
+        // }
+        //
+        return this.parseOrder (response);
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // TODO : is this https://docs.woo.org/#get-orders correclty implemented by me? Maybe I missed some things (like end-time handing?)
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        if (since !== undefined) {
+            request['start_t'] = since;
+        }
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchOrders', market, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PrivateGetOrders',
+        });
+        const response = await this[method] (this.extend (request, query));
+        const data = this.safeValue (response, 'rows');
+        return this.parseOrders (data, market, since, limit, params);
+    }
+
+    parseOrder (order, market = undefined) {
+        //
+        // Possible input functions:
+        // * createOrder
+        // * cancelOrder
+        // * fetchOrder
+        // * fetchOrders
+        // const isFromFetchOrder = ('order_tag' in order); TO_DO
+        const timestamp = this.safeTimestamp2 (order, 'timestamp', 'created_time');
+        const orderId = this.safeInteger (order, 'order_id');
+        const clientOrderId = this.safeTimestamp (order, 'client_order_id'); // Somehow, this always returns 0 for limit order
+        if (market === undefined) {
+            const marketId = this.safeString (order, 'symbol');
+            market = this.safeMarket (marketId, market);
+        }
```

Comment:
The `if (market === undefined)` conditional is not necessary here – it is done in safeMarket. Let's replace the four lines 865-868 with just these two lines:

```JavaScript
            const marketId = this.safeString (order, 'symbol');
            market = this.safeMarket (marketId, market);
```